File: space-game/Core.Tests/Data/SampleTextures.vb
Imports Core.Classes
Imports Core.Enums

Namespace Data
    Public Class SampleTextures
        Public ReadOnly High As Texture
        Public ReadOnly Medium As Texture
        Public ReadOnly Low As Texture
        Public ReadOnly EmptyHigh As Texture
        Public ReadOnly EmptyMedium As Texture
        Public ReadOnly EmptyLow As Texture
        Public ReadOnly Type As TextureType

        Public Sub New(type As TextureType)
            Type = type

            Low = New Texture() With {.QualityEnum = TextureQuality.Low,
                                            .TypeEnum = Type,
                                            .Path = "low texture path"}

            Medium = New Texture() With {.QualityEnum = TextureQuality.Medium,
                                            .TypeEnum = Type,
                                            .Path = "medium texture path"}

            High = New Texture() With {.QualityEnum = TextureQuality.High,
                                            .TypeEnum = Type,
                                            .Path = "high texture path"}


            EmptyLow = New Texture() With {.QualityEnum = TextureQuality.Low,
                                            .TypeEnum = Type,
                                            .Path = " "}

            EmptyMedium = New Texture() With {.QualityEnum = TextureQuality.Medium,
                                            .TypeEnum = Type,
                                            .Path = " "}

            EmptyHigh = New Texture() With {.QualityEnum = TextureQuality.High,
                                            .TypeEnum = Type,
                                            .Path = " "}
        End Sub
    End Class
End Namespace

File: space-game/Core.Tests/Data/SampleTextures.cs
using Core.Classes;
using Core.Enums;

namespace Core.Tests.Data
{
    public class SampleTextures
    {
        public readonly Texture High;
        public readonly Texture Medium;
        public readonly Texture Low;
        public readonly Texture EmptyHigh;
        public readonly Texture EmptyMedium;
        public readonly Texture EmptyLow;

        public readonly TextureType Type;
        public SampleTextures(TextureType type)
        {
            Type = type;

            Low = new Texture
            {
                QualityEnum = TextureQuality.Low,
                TypeEnum = type,
                Path = "low texture path"
            };

            Medium = new Texture
            {
                QualityEnum = TextureQuality.Medium,
                TypeEnum = type,
                Path = "medium texture path"
            };

            High = new Texture
            {
                QualityEnum = TextureQuality.High,
                TypeEnum = type,
                Path = "high texture path"
            };


            EmptyLow = new Texture
            {
                QualityEnum = TextureQuality.Low,
                TypeEnum = type,
                Path = " "
            };

            EmptyMedium = new Texture
            {
                QualityEnum = TextureQuality.Medium,
                TypeEnum = type,
                Path = " "
            };

            EmptyHigh = new Texture
            {
                QualityEnum = TextureQuality.High,
                TypeEnum = type,
                Path = " "
            };
        }
    }
}
